test/Microsoft.ML.Tests/TrainerEstimators/MatrixFactorizationTests.cs

```diff
@@ -91,15 +94,72 @@ public void MatrixFactorizationSimpleTrainAndPredict()
                 var mlContext = new MLContext();
                 var metrices = mlContext.Regression.Evaluate(prediction, label: labelColumnName, score: scoreColumnName);
 
-                // Determine if the selected metric is reasonable for differen
-                var expectedWindowsL2Error = 0.61528733643754685; // Windows baseline
-                var expectedMacL2Error = 0.61192207960271; // Mac baseline
-                var expectedLinuxL2Error = 0.616821448679879; // Linux baseline
-                double tolerance = System.Math.Pow(10, -DigitsOfPrecision);
-                bool inWindowsRange = expectedWindowsL2Error - tolerance < metrices.L2 && metrices.L2 < expectedWindowsL2Error + tolerance;
-                bool inMacRange = expectedMacL2Error - tolerance < metrices.L2 && metrices.L2 < expectedMacL2Error + tolerance;
-                bool inLinuxRange = expectedLinuxL2Error - tolerance < metrices.L2 && metrices.L2 < expectedLinuxL2Error + tolerance;
-                Assert.True(inWindowsRange || inMacRange || inLinuxRange);
+                // Determine if the selected metric is reasonable for different platforms
+                double tolerance = Math.Pow(10, -6); // -7 sometime leads to a test failure on Mac
+                if (Environment.OSVersion.Platform == PlatformID.Unix)
+                {
+                    // Unix case
+                    var expectedUnixL2Error = 0.616821448679879; // Unix baseline
+                    Assert.InRange(metrices.L2, expectedUnixL2Error - tolerance, expectedUnixL2Error + tolerance);
+                }
+                else if (Environment.OSVersion.Platform == PlatformID.MacOSX)
+                {
+                    // Mac case
+                    var expectedMacL2Error = 0.61192207960271; // Mac baseline
+                    Assert.InRange(metrices.L2, expectedMacL2Error - tolerance, expectedMacL2Error + tolerance);
+                }
+                else
+                {
+                    // Windows case
+                    var expectedWindowsL2Error = 0.61528733643754685; // Windows baseline
+                    Assert.InRange(metrices.L2, expectedWindowsL2Error - tolerance, expectedWindowsL2Error + tolerance);
+                }
+            }
+        }
+
+        [Fact]
+        public void MatrixFactorizationStatic()
+        {
+            using (var env = new LocalEnvironment(seed: 1, conc: 1))
```

Comment:
Let's use `MLContext` in our new tests instead. Then you don't need line 127 either.

Reply:
Ok.